app/src/org/commcare/preferences/DeveloperPreferences.java

```diff
@@ -94,14 +97,31 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         GoogleAnalyticsUtils.reportPrefActivityEntry(GoogleAnalyticsFields.CATEGORY_DEV_PREFS);
         populatePrefKeyToEventLabelMapping();
-        initAllPrefs();
+
+        setTitle("Developer Options");
+        getPreferenceManager().setSharedPreferencesName(
+                CommCareApplication.instance().getCurrentApp().getPreferencesFilename());
+
+        if (userAccessPinNeeded()) {
+            addPreferencesFromResource(R.xml.preferences_developer_pin);
+        } else {
+            initAllPrefs();
+        }
+    }
+
+    private static boolean userAccessPinNeeded() {
+        SharedPreferences prefs = CommCareApplication.instance().getCurrentApp().getAppPreferences();
```

Comment:
@amstone326 We should set this to not require a pin if the user is a SuperAdmin (or whatever the barcode scan/APK_IN_DEBUG check is) as well, so we have a way to access developer options without needing project specific data.